test/test_typing.py

```diff
@@ -0,0 +1,150 @@
+# based on NumPy numpy/typing/tests/test_typing.py
+
+import itertools
+import os
+import re
+import shutil
+from typing import IO, Dict, List
+
+import pytest
+import numpy as np
+
+try:
+    from mypy import api
+except ImportError:
+    NO_MYPY = True
+else:
+    NO_MYPY = False
+
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), "typing")
+REVEAL_DIR = os.path.join(DATA_DIR, "reveal")
+MYPY_INI = os.path.join(DATA_DIR, os.pardir, os.pardir, "mypy.ini")
+CACHE_DIR = os.path.join(DATA_DIR, ".mypy_cache")
+
+#: A dictionary with file names as keys and lists of the mypy stdout as values.
+#: To-be populated by `run_mypy`.
+OUTPUT_MYPY: Dict[str, List[str]] = {}
+
+
+def _key_func(key: str) -> str:
+    """Split at the first occurance of the ``:`` character.
+
+    Windows drive-letters (*e.g.* ``C:``) are ignored herein.
+    """
+    drive, tail = os.path.splitdrive(key)
+    return os.path.join(drive, tail.split(":", 1)[0])
+
+
+@pytest.mark.skipif(NO_MYPY, reason="Mypy is not installed")
+@pytest.fixture(scope="module", autouse=True)
+def run_mypy() -> None:
+    """Clears the cache and run mypy before running any of the typing tests.
+
+    The mypy results are cached in `OUTPUT_MYPY` for further use.
+
+    """
+    if os.path.isdir(CACHE_DIR):
+        shutil.rmtree(CACHE_DIR)
+
+    for directory in (REVEAL_DIR,):
+        # Run mypy
+        stdout, stderr, _ = api.run(
+            [
+                "--show-absolute-path",
+                "--config-file",
+                MYPY_INI,
+                "--cache-dir",
+                CACHE_DIR,
+                directory,
+            ]
+        )
+        assert not stderr, directory
+        stdout = stdout.replace("*", "")
+
+        # Parse the output
+        iterator = itertools.groupby(stdout.split("\n"), key=_key_func)
+        OUTPUT_MYPY.update((k, list(v)) for k, v in iterator if k)
+
+
+def get_test_cases(directory):
+    for root, _, files in os.walk(directory):
+        for fname in files:
+            if os.path.splitext(fname)[-1] == ".py":
+                fullpath = os.path.join(root, fname)
+                # Use relative path for nice py.test name
+                relpath = os.path.relpath(fullpath, start=directory)
+
+                yield pytest.param(
+                    fullpath,
+                    # Manually specify a name for the test
+                    id=relpath,
+                )
+
+
+def _construct_format_dict():
+    return {}
+
+
+#: A dictionary with all supported format keys (as keys)
+#: and matching values
+FORMAT_DICT: Dict[str, str] = _construct_format_dict()
```

Comment:
Sorry if I miss some context in haste, but why do you prefer to have a function rather than initialize dictionary in place? (And if you want to have function, why not type-annotate it (i.e. `def _construct_format_dict()-> Dict[str, str
`)  and let type-checker infer the type of the variable)
```suggestion
#: A dictionary with all supported format keys (as keys)
#: and matching values
FORMAT_DICT: Dict[str, str] = {}
```

Reply:
This was inherited from numpy source code. I'll change the code to your suggestion